### PR TITLE
refactor(Telemetry): Use `prompt-with-history`

### PR DESCRIPTION
**Pull request by pgrzesik**
Use `promptWithHistory` that will automatically record questions in step history. There is one exception where we still use `inquirer.prompt` directly, but it was easier to make an exception here in my opinion as we have to access original prompt from inquirer than to change the implementation just for this case. 

Before merging this PR, corresponding backend PRs have to be merged and deployed.

Addresses: #9367

### CODE REVIEW

File: lib/cli/interactive-setup/dashboard-set-org.js [hunk 1]
```diff
@@ -13,92 +14,84 @@ const {
...
-        {
-          name: 'Switch to one of the existing apps (or create new one with different name)',
-          value: 'chooseExisting',
-        },
-        { name: "Skip, I'll sort this out manually", value: 'skip' },
-      ],
-    })
-  ).appUpdateType;
-  stepHistory.set('appUpdateType', appUpdateType);
+  // TODO: FIX how the value is recorded
```
Comment (medikoo) [inline lib/cli/interactive-setup/dashboard-set-org.js:35]: What exactly is the problem?

File: test/unit/lib/cli/interactive-setup/aws-credentials.test.js [hunk 2]
```diff
@@ -348,10 +348,10 @@ describe('test/unit/lib/cli/interactive-setup/aws-credentials.test.js', () => {
       expect(context.stepHistory.valuesMap()).to.deep.equal(
         new Map([
           ['credentialsSetupChoice', '_local_'],
-          ['createAwsAccountPrompt', true],
-          ['generateAwsCredsPrompt', true],
-          ['accessKeyId', '_user_provided_'],
-          ['secretAccessKey', '_user_provided_'],
+          ['createAwsAccountPrompt', '_user_input_'],
+          ['generateAwsCredsPrompt', '_user_input_'],
```
Comment (medikoo) [inline test/unit/lib/cli/interactive-setup/aws-credentials.test.js:352]: This looks confusing. We dedicated `_user_input_` to user's free form text input, while here is definitely not a case.

Can't we resolve it to something as `_continuation_` ?

**Review by medikoo — COMMENTED** [review]
Looks very good I have just one concern towards one prompt type, otherwise it looks perfect to me.

File: lib/cli/interactive-setup/dashboard-set-org.js (hunk 1 above)
Comment (pgrzesik) [reply lib/cli/interactive-setup/dashboard-set-org.js:35]: oops, that's a leftover todo - it's all good here

File: test/unit/lib/cli/interactive-setup/aws-credentials.test.js (hunk 2 above)
Comment (pgrzesik) [reply test/unit/lib/cli/interactive-setup/aws-credentials.test.js:352]: I guess we can - the actual answer that is being record is empty string, so we could default to `_continuation_` in such cases or just record an empty string - what do you think would be better. I like the `_continuation_` suggestion

Comment (medikoo) [reply test/unit/lib/cli/interactive-setup/aws-credentials.test.js:352]: Yes, I'd stick to `_continuation_` (nothing better comes to my mind)

**Review by medikoo — APPROVED** [review]
Looks great 👍

**Pull request merged**